modules/teadsBidAdapter.js

```diff
@@ -230,6 +232,16 @@ function getConnectionDownLink(nav) {
   return nav && nav.connection && nav.connection.downlink >= 0 ? nav.connection.downlink.toString() : '';
 }
 
+function getNetworkQuality(navigator) {
+  const connection = navigator.connection || navigator.mozConnection || navigator.webkitConnection;
+
+  return connection?.effectiveType ?? '';
+}
+
+function getDomComplexity(document) {
+  return document?.querySelectorAll('*')?.length ?? -1;
```

Comment:
I don't think the second `?.` is necessary because if `document` exists then `querySelectorAll` will always return a [NodeList](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll#return_value). 
```suggestion
  return document?.querySelectorAll('*').length ?? -1;
```

Same change for the test

Reply:
It's more out of safety than anything, because having the `?` doesn't have any performance impact whatsoever and it serves as error-proofing if ever the Web API were to change.